rust/xaynet-server/src/storage/redis/mod.rs

```diff
@@ -193,40 +185,41 @@ impl Connection {
                 PublicEncryptKeyWrite::from(ephm_pk),
             )
             .await
+            .map_err(to_storage_err)
     }
 
-    /// Gets the [`SeedDict`] entry for the given ['SumParticipantPublicKey'] or an empty map
-    /// when a [`SeedDict`] entry does not exist.
-    pub async fn get_seed_dict_for_sum_pk(
-        mut self,
-        sum_pk: &SumParticipantPublicKey,
-    ) -> RedisResult<HashMap<UpdateParticipantPublicKey, EncryptedMaskSeed>> {
-        debug!(
-            "get seed dictionary for sum participant with pk {:?}",
-            sum_pk
-        );
+    async fn sum_dict(&mut self) -> StorageResult<Option<SumDict>> {
+        debug!("get sum dictionary");
         // https://redis.io/commands/hgetall
         // > Return value
         //   Array reply: list of fields and their values stored in the hash, or an empty
         //   list when key does not exist.
-        let result: Vec<(PublicSigningKeyRead, EncryptedMaskSeedRead)> = self
+        let reply: Vec<(PublicSigningKeyRead, PublicEncryptKeyRead)> = self
             .connection
-            .hgetall(PublicSigningKeyWrite::from(sum_pk))
-            .await?;
-        let seed_dict = result
-            .into_iter()
-            .map(|(pk, seed)| (pk.into(), seed.into()))
-            .collect();
+            .hgetall("sum_dict")
+            .await
+            .map_err(to_storage_err)?;
 
-        Ok(seed_dict)
+        let result = match reply.is_empty() {
```

Comment:
This is personal taste but I think an early return is more explicit here:

```rust
        if reply.is_empty() {
            return Ok(None);
        }

        let sum_dict = reply
            .into_iter()
            .map(|(pk, ephm_pk)| (pk.into(), ephm_pk.into()))
            .collect();

        Ok(Some(sum_dict))
```

Reply:
fixed